cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/EquipmentExport.java

```diff
@@ -365,7 +367,15 @@ private static void writeGenerators(Network network, Map<Terminal, String> mapTe
         Set<String> generatingUnitsWritten = new HashSet<>();
         for (Generator generator : network.getGenerators()) {
             String cgmesOriginalClass = generator.getProperty(Conversion.PROPERTY_CGMES_ORIGINAL_CLASS, CgmesNames.SYNCHRONOUS_MACHINE);
-
+            Terminal regulatingTerminal = generator.getRegulatingTerminal();
+            String mode = RegulatingControlEq.REGULATING_CONTROL_VOLTAGE;
+
+            // Not possible to re-export local reactive power control.
+            RemoteReactivePowerControl rrpc = generator.getExtension(RemoteReactivePowerControl.class);
+            if (rrpc != null) {
+                regulatingTerminal = rrpc.getRegulatingTerminal();
+                mode = RegulatingControlEq.REGULATING_CONTROL_REACTIVE_POWER;
+            }
             switch (cgmesOriginalClass) {
```

Comment:
My proposal:

```
RemoteReactivePowerControl rrpc = generator.getExtension(RemoteReactivePowerControl.class);
String mode = getGeneratorRegulatingControlMode(generator, rrpc);
Terminal regulatingTerminal = mode.equals(RegulatingControlEq.REGULATING_CONTROL_VOLTAGE) ? generator.getRegulatingTerminal() : rrpc.getRegulatingTerminal();
```

```
private static String getGeneratorRegulatingControlMode(Generator generator, RemoteReactivePowerControl rrpc) {
        if (rrpc == null) {
            return RegulatingControlEq.REGULATING_CONTROL_VOLTAGE;
        }
        boolean enabledVoltageControl = generator.isVoltageRegulatorOn();
        boolean enabledReactivePowerControl = rrpc.isEnabled();
        
        if (enabledVoltageControl) {
            return RegulatingControlEq.REGULATING_CONTROL_VOLTAGE;
        } else if (enabledReactivePowerControl) {
            return RegulatingControlEq.REGULATING_CONTROL_REACTIVE_POWER;
        } else {
            boolean validVoltageSetpoint = SteadyStateHypothesisExport.isValidVoltageSetpoint(generator.getTargetV());
            if (validVoltageSetpoint) {
                return RegulatingControlEq.REGULATING_CONTROL_VOLTAGE;
            } else {
                return RegulatingControlEq.REGULATING_CONTROL_REACTIVE_POWER;
            }
        }
    }
```
We can define the `getGeneratorRegulatingControlMode` method in the `CgmesExportUtil` class and use it during the `SSH` export to be coherent with the `EQ` export.

Reply:
The difficulty here is in priorities: Is the RRPC extension priority or voltage control?

Example: rrpc exists, voltage control is enabled -> We export voltage control, regardless of rrpc status.

Since rrpc  has to be created explicitly as an extension, assumption is that it has the priority.
Therefore, if rrpc is there, we ignore voltage control completely and only use rrpc parameters.

Reply:
If the reactive power control extension is present and enabled, it does not override the voltage control if enabled. Maybe it is a mistake but it is what is developed inside open loadflow.

Reply:
I prefer to select the control priority using the electrical information rather than the  software information. The final user, in most of the cases, will not know how the model is internally implemented. Then, if both controls have the same valid attributes I prefer to export the voltage control. But as before, both options are valid.